tests/test_constants_speed_of_sound.py

```diff
@@ -34,3 +34,107 @@ def test_speed_of_sound_simple_edge_cases():
     expected = 343.2 * np.sqrt((np.array(temperatures) + 273.15) / 293.15)
     result = pf.constants.speed_of_sound_simple(temperatures)
     npt.assert_array_equal(result, expected)
+
+
+def test_speed_of_sound_ideal_gas_typical_values():
+    temperature = 20.0  # Celsius
+    relative_humidity = 0.5  # 50%
+    atmospheric_pressure = 101325  # Pa
+    result = pf.constants.speed_of_sound_ideal_gas(
+        temperature, relative_humidity, atmospheric_pressure)
+    expected = 343.2*np.sqrt((temperature+273.15)/293.15)
+    assert np.isclose(result, expected, rtol=1e-2)
+
+
+@pytest.mark.parametrize("temperature", [  # Celsius
+    [20, 20.0],
+    np.array([20, 20.0]),
+    (20, 20.0),
+    20,
+])
+@pytest.mark.parametrize("relative_humidity", [
+    [.5, .5],
+    np.array([.5, .5]),
+    (.5, .5),
+    .5,
+])
+@pytest.mark.parametrize("atmospheric_pressure", [  # Pa
+    [101325, 101325],
+    np.array([101325, 101325]),
+    (101325, 101325),
+    101325,
+])
+def test_speed_of_sound_ideal_gas_array_like(
+        temperature, relative_humidity, atmospheric_pressure):
+    result = pf.constants.speed_of_sound_ideal_gas(
+        temperature, relative_humidity, atmospheric_pressure)
+    npt.assert_almost_equal(result, 343.8278399)
+
+
+def test_speed_of_sound_ideal_gas_values_p_water():
+    temperature = 20.0  # Celsius
+    relative_humidity = 0.5  # 50%
+    atmospheric_pressure = 101325  # Pa
+    p_water = pf.constants.saturation_vapor_pressure(temperature)
+    result = pf.constants.speed_of_sound_ideal_gas(
+        temperature, relative_humidity, atmospheric_pressure, p_water)
+    expected = pf.constants.speed_of_sound_ideal_gas(
+        temperature, relative_humidity, atmospheric_pressure)
+    assert np.isclose(result, expected)
+
+
+def test_speed_of_sound_ideal_gas_edge_temperature():
+    temperature = -20.0  # Celsius
+    relative_humidity = 0.5  # 50%
+    atmospheric_pressure = 101325  # Pa
+    result = pf.constants.speed_of_sound_ideal_gas(
+        temperature, relative_humidity, atmospheric_pressure)
+    expected = 343.2*np.sqrt((temperature+273.15)/293.15)
```

Comment:
This is used quite often. Maybe put it into a private function and briefly explain where it is coming from to make the tests easier to understand